### PR TITLE
[WIP] Remove the sample tracking functionality

### PR DESCRIPTION
**Pull request by guerler**
This PR removes the sample request handling. It is branched from #4480 and will be split into smaller, more digestible changesets.

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/buildapp.py [hunk 1]
```diff
@@ -105,10 +105,11 @@ def paste_app_factory(global_conf, **kwargs):
 
     webapp.add_client_route('/admin/users', 'admin')
     webapp.add_client_route('/admin/roles', 'admin')
+    webapp.add_client_route('/admin/forms', 'admin')
     webapp.add_client_route('/admin/groups', 'admin')
     webapp.add_client_route('/admin/tool_versions', 'admin')
     webapp.add_client_route('/admin/quotas', 'admin')
-    webapp.add_client_route('/admin/forms/{form_id}', 'admin')
+    webapp.add_client_route('/admin/form/{form_id}', 'admin')
```
Comment (martenson) [inline lib/galaxy/webapps/galaxy/buildapp.py]: why is this being changed?

Comment (guerler) [reply lib/galaxy/webapps/galaxy/buildapp.py]: This PR is branched from #4480. I updated the description above. We are changing this line in that branched PR to keep the resource url consistent with the actual `forms` resource.

File: lib/galaxy/model/__init__.py [hunk 2]
```diff
@@ -4289,12 +4171,7 @@ def file_name(self):
 class FormDefinition(object, Dictifiable):
     # The following form_builder classes are supported by the FormDefinition class.
     supported_field_types = [AddressField, CheckboxField, PasswordField, SelectField, TextArea, TextField, WorkflowField, WorkflowMappingField, HistoryField]
-    types = Bunch(REQUEST='Sequencing Request Form',
-                  SAMPLE='Sequencing Sample Form',
-                  EXTERNAL_SERVICE='External Service Information Form',
-                  RUN_DETAILS_TEMPLATE='Sample run details template',
-                  LIBRARY_INFO_TEMPLATE='Library information template',
```
Comment (martenson) [inline lib/galaxy/model/__init__.py:4263]: This removal will make this PR wait for the admin library UI removal, which should come before 18.01 but later than the rest of this PR.

**Comment by guerler** [comment]
Closing this in favor of #4526.

**Comment by dannon** [comment]
@guerler Was this meant to be reopened?  It looks like mostly library removal commits?

**Comment by guerler** [comment]
Yeah, it is branched from the library removal. Once that one is merged this PR should be much smaller.

**Comment by dannon** [comment]
Ok.  Maybe it's just me, but the branched PRs are a whole lot harder to digest and personally I think it'd make it a lot easier to parse if they were used sparingly.  Admittedly, it's hard to weed through all the changes here to see what's actually dependent work and what isn't, but this looks like a situation where 'standard' conflict resolution would have worked fine and there wasn't a need to chain everything together.

It also looks like all the *last* commits on this branch are library stuff; not the first stuff (as I'd expect if this was branched off that).  Is that expected?

**Pull request closed without merging**